mohawk/base.py

```diff
@@ -31,6 +31,13 @@ def _authorize(self, mac_type, parsed_header, resource,
 
         now = utc_now(offset_in_seconds=localtime_offset_in_seconds)
 
+        their_hash = parsed_header.get('hash', '')
+        mac = calculate_mac(mac_type, resource, their_hash)
+        if not strings_match(mac, parsed_header['mac']):
```

Comment:
I know you just moved `parsed_header['mac']` from elsewhere but I can't see any way that this key is guaranteed to exist. I think it should be changed to `parsed_header.get('mac')` and there is one other instance to change.